src/lib/QtWidgets/QBoxLayout.ts

```diff
@@ -0,0 +1,49 @@
+import addon from '../utils/addon';
+import { NodeWidget } from './QWidget';
+import { NodeLayout } from './QLayout';
+import { NativeElement } from '../core/Component';
+import { Direction } from '../QtEnums';
+
+export class QBoxLayout extends NodeLayout {
+    native: NativeElement;
+    constructor(dir: Direction, parent?: NodeWidget) {
+        let native: NativeElement;
+        if (parent) {
+            native = new addon.QBoxLayout(dir, parent.native);
+        } else {
+            native = new addon.QBoxLayout(dir);
+        }
+        super(native);
+        this.nodeParent = parent;
+        this.native = native;
+    }
+    addLayout(layout: NodeLayout, stretch = 0): void {
+        this.native.addLayout(layout.native, stretch);
```

Comment:
Overall fantastic job! 
We may need to save this layout in a member variable of this QBoxLayout class othewise it might get garbage collected by js after sometime.